plan/physical_plan_builder.go

```diff
@@ -61,6 +62,14 @@ func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInf
 		ts.readOnly = true
 	}
 
+	infoSchema, ok := getInfoSchema(p.ctx)
```

Comment:
I'm afraid `getInfoSchema` here would break the contract and can't ensure results' correctness.

In our prerequisite:

schema information is got before building a plan, and plan builder should always use that schema -- `p.Schema()`

If schema is changed during plan building process, the generated executor would fail to commit (schema validator check fail), we will detect that and rebuild plan.